lib/adf_diag.py

```diff
@@ -1080,3 +1108,227 @@ def derive_variables(self, vars_to_derive=None, ts_dir=None, overwrite=None):
+                    hist_file_ds = xr.open_dataset(
+                        adf_file, decode_cf=False, decode_times=False
+                    )
+                    if "time_period_freq" in hist_file_ds.attrs:
+                        dataset_freq = hist_file_ds.attrs["time_period_freq"]
+                        if verbose > 2:
+                            print(f"time_period_freq attribute found: {dataset_freq}")
+                    else:
+                        if verbose > 0:
+                            print(
+                                f"ERROR: Necessary 'time_period_freq' attribute missing from {adf_file}"
+                            )
+                        continue
+
+                    found_strings = [
+                        word for word in freq_string_options if word in dataset_freq
+                    ]
+                    if len(found_strings) == 1:
+                        if verbose > 2:
+                            print(
+                                f"Found dataset_freq {dataset_freq} matches {found_strings}"
+                            )
+                    elif len(found_strings) > 1:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: Found dataset_freq {dataset_freq} matches multiple string possibilities:{', '.join(found_strings)}"
+                            )
+                    else:
+                        if verbose > 0:
+                            print(
+                                "ERROR: None of the frequency options {freq_string_options} are present in the time_period_freq attribute {time_perido_freq}"
+                            )
+                            print("Skipping {adf_file}")
+                            freq = "frequency_missing"
+                        continue
+                    freq = found_strings[0]
+
+                    #
+                    # Destination file is MDTF directory and name structure
+                    #
+                    mdtf_dir = os.sep.join([mdtf_model_data_root, case_name, freq])
+
+                    os.makedirs(mdtf_dir, exist_ok=True)
+                    mdtf_file = (
+                        mdtf_dir + os.sep + ".".join([case_name, var, freq, "nc"])
+                    )
+                    mdtf_file_list = glob.glob(
+                        mdtf_file
+                    )  # Check if files already exist in MDTF directory
+                    if (
+                        mdtf_file_list
+                    ):  # If files exist, then should check if over-writing is allowed:
+                        if verbose > 0:
+                            print(
+                                f"\t   INFO: not clobbering existing mdtf file {mdtf_file_list}"
+                            )
+                        # logic to allow override#  if not overwrite_mdtf[case_idx]:
```

Comment:
Is there supposed to be overwrite logic here, or is this a future effort?

Reply:
Am I wrong that there *is* overwrite logic?
if the file is present
then continue 
else copy file

> mdtf_file_list = glob.glob(mdtf_file)  # Check if files already exist in MDTF directory
                    if (mdtf_file_list):  # If files exist, then should check if over-writing is allowed:
                        if verbose > 0:
                            print(
                                f"\t   INFO: not clobbering existing mdtf file {mdtf_file_list}"
                            )
                        # logic to allow override#  if not overwrite_mdtf[case_idx]:
                        continue  # simply skip file copy for this variable:

Reply:
Oh, I see where I was misunderstanding. Yes, there should be logic saying over-write or don't. At the moment, the user can just remove the files if they want them re-copied.